amx/server/syscalls.lua

```diff
@@ -2942,6 +2942,42 @@ function GetVehicleParamsSirenState(amx, vehicle)
 		return 0
 	end
 end
+
+
+-- Weapon
+
+-- This part looks like a dirt
+function _PutReloadingStatus(status)
+	g_Players[getElemID(client)].reloading = status
+end
+addEvent("onPuttingReloadingStatus", true)
+addEventHandler("onPuttingReloadingStatus", resourceRoot, _PutReloadingStatus)
+
+function GetPlayerWeaponState(amx, player)
+	-- -1 WEAPONSTATE_UNKNOWN 
+	-- 0 WEAPONSTATE_NO_BULLETS
+	-- 1 WEAPONSTATE_LAST_BULLET
+	-- 2 WEAPONSTATE_MORE_BULLETS
+	-- 3 WEAPONSTATE_RELOADING
+
+	local vehicle = getPedOccupiedVehicle(player)
+	if vehicle ~= nil then return -1 end
+
+	-- Someone tell me how can I transfer the value of this vars to the server?
+	clientCall(player, '_isPedReloadingWeapon', player)

```

Comment:
The way it would work is that there's an event client-side like `onClientPedReload` -- and in that event, sync it to the server. Did you test whether this would work?

I think this function will always return the result of the previous call.

My suggestion is that you just never `return 3`, and when we implement stuff in MTA, we can add the `return 3`.

Reply:
> I think we should just try to implement https://wiki.multitheftauto.com/wiki/IsPedReloadingWeapon server-side in MTA

I'll be wait it with pleasure!

Reply:
> 
> 
> The way it would work is that there's an event client-side like `onClientPedReload` -- and in that event, sync it to the server. Did you test whether this would work?

The part of the code with reloading a gun is working. I have tested it. (You can see it at my screenshot)

Reply:
There's no guarantee that the reloading state will be updated between line 2967 and 2968. The event call involves networking with the client and is an asynchronous operation.